bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/transceiver/EnOceanTransceiver.java

```diff
@@ -313,13 +313,19 @@ protected int read(byte[] buffer, int length) {
         InputStream localInputStream = inputStream;
         if (localInputStream != null) {
             try {
-                localInputStream.read(buffer, 0, length);
+                return localInputStream.read(buffer, 0, length);
             } catch (IOException e) {
                 logger.debug("IOException occured while reading the input stream", e);
                 return 0;
             }
+        } else {
+            logger.error("Cannot read from null stream");
+            Future<?> localReadingTask = readingTask;
+            if (localReadingTask != null) {
+                localReadingTask.cancel(true);
```

Comment:
Should `readingTask` be set to null after this?

Reply:
It shouldn't be necessary because when (re)creating, test is also done with " ! isCancelled"
But better safe than sorry, I guess there is no harm in doing it.
So, done !